celery/exceptions.py

```diff
@@ -144,17 +153,18 @@ class Retry(TaskPredicate):
     """The task is to be retried later."""
 
     #: Optional message describing context of retry.
-    message = None
+    message: Optional[str] = None
 
     #: Exception (if any) that caused the retry to happen.
-    exc = None
+    exc: Optional[Exception] = None
+    excs: Optional[str] = None
 
-    #: Time of retry (ETA), either :class:`numbers.Real` or
-    #: :class:`~datetime.datetime`.
-    when = None
+    #: Time of retry (ETA)
```

Comment:
```suggestion
    #: Time of retry (ETA), either :class:`numbers.Number` or
    #: :class:`~datetime.datetime`.
```
@Nusnus  I didn't get what you said. Is this enough?

Reply:
:class:numbers.Real is what he suggested as per my understanding

Reply:
So... A mismatch between doc and code? 🧐

Reply:
TBH you're right. My original thought was that the doc would mention Real and the annotation would mention Number, being _upper in hierarchy_, but that is wrong as you are noting @Kludex  :)